lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -111,38 +84,98 @@ describe('NxTabs', function () {
+  });
+
+  it('calls onTabSelect when a tab is clicked', async function() {
+    const user = userEvent.setup();
+    const onTabSelect = jest.fn();
+
+    const { getByRole } = quickRender({
+      activeTab: -1,
+      onTabSelect
+    });
+
+    await user.click(getByRole('tab'));
+
+    expect(onTabSelect).toHaveBeenCalledWith(0);
   });
 
-  it('calls onTabSelect when a tab is clicked', function () {
+  it('calls onTabSelect when a tab is selected via keyboard', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+
+    const { getAllByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect,
+      children: (
         <NxTabList>
           <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
         </NxTabList>
-      </NxTabs>
-    );
+      )
+    });
 
-    component.find('[role="tab"]').simulate('click');
+    const firstTab = getAllByRole('tab')[0];
 
-    expect(onTabSelect).toHaveBeenCalledWith(0);
+    firstTab.focus();
+    await user.keyboard('{ArrowRight}{Enter}');
+
+    expect(onTabSelect).toHaveBeenCalledWith(1);
   });
 
-  it('calls onTabSelect when a tab is selected via keyboard', function () {
+  it('calls onTabSelect when a tab is selected via keyboard', async function() {
+    const user = userEvent.setup();
     const onTabSelect = jest.fn();
-    const component = mount(
-      <NxTabs activeTab={0} onTabSelect={onTabSelect}>
+
+    const { getAllByRole } = quickRender({
+      activeTab: 0,
+      onTabSelect,
+      children: (
         <NxTabList>
           <NxTab>Tab 0</NxTab>
+          <NxTab>Tab 1</NxTab>
         </NxTabList>
-      </NxTabs>
-    );
+      )
+    });
 
-    component.find('[role="tab"]').simulate('keypress', { key: ' ' });
+    const firstTab = getAllByRole('tab')[0];
 
-    expect(onTabSelect).toHaveBeenCalledWith(0);
+    firstTab.focus();
+    await user.keyboard('{ArrowRight}{Enter}');
+
+    expect(onTabSelect).toHaveBeenCalledWith(1);
+  });
+
+  it('sets id and aria labelledby to panel to reference active tab aria-controls', async function() {
+    const user = userEvent.setup();
+
+    const Fixture = () => {
+      const [activeTab, setActiveTab] = useState<number>(0);
+      return (
+        <NxTabs activeTab={activeTab} onTabSelect={setActiveTab}>
+          <NxTabList>
+            <NxTab>Tab 0</NxTab>
+            <NxTab>Tab 1</NxTab>
+          </NxTabList>
+          <NxTabPanel>Content 0</NxTabPanel>
+          <NxTabPanel>Content 1</NxTabPanel>
+        </NxTabs>
+      );
+    };
+
+    const { getByRole } = render(<Fixture />);
+    const getActiveTabControls = () => getByRole('tab', { selected: true }).getAttribute('aria-controls');
+
+    expect(getByRole('tabpanel').id).toEqual(getActiveTabControls());
+    expect(getByRole('tabpanel')).toHaveAccessibleName('Tab 0 Tab 0');
```

Comment:
I noticed that the hidden part is part of the accessible name.
Do you think we should add `aria-hidden` to that?

Reply:
> I noticed that the hidden part is part of the accessible name.
Do you think we should add aria-hidden to that?

Check to see if it's part of the accessible name in the actual browser - it might be different there due to the inclusion of the CSS.  If it's part of the accessible name there too though, then yeah we should fix that.

On the other hand if this is just an issue in the unit tests (due to the lack of CSS) I'd work around it by using a regex for the a11y name so that it doesn't have to be an exact, full match

Reply:
> > I noticed that the hidden part is part of the accessible name.
> > Do you think we should add aria-hidden to that?
> 
> Check to see if it's part of the accessible name in the actual browser - it might be different there due to the inclusion of the CSS. If it's part of the accessible name there too though, then yeah we should fix that.
> 
> On the other hand if this is just an issue in the unit tests (due to the lack of CSS) I'd work around it by using a regex for the a11y name so that it doesn't have to be an exact, full match

Thanks, it looks okay on the browser:
I've updated it to use regex here:
10c589cc6b4ec332aa20394fd07bf3f83089954a